lib/features/steps/request_assertion_steps.rb

```diff
@@ -33,7 +43,20 @@ def find_request(request_index)
 
 Then(/^the "(.+)" header is a timestamp(?: for request (\d+))?$/) do |header_name, request_index|
   header = find_request(request_index)[:request][header_name]
-  assert_match(/^\d{4}\-\d{2}\-\d{2}T\d{2}:\d{2}:[\d\.]+Z?$/, header)
+  assert_match(timestamp_regex, header)
+end
+
+Then(/^the "(.+)" query parameter equals "(.+)"(?: for request (\d+))?$/) do |parameter_name, parameter_value, request_index|
+  assert_equal(parameter_value, parse_querystring(find_request(request_index))[parameter_name][0])
+end
+
+Then(/^the "(.+)" query parameter is not null(?: for request (\d+))?$/) do |parameter_name, request_index|
+  assert_not_nil(parse_querystring(find_request(request_index))[parameter_name][0], "The '#{parameter_name}' query parameter should not be null")
+end
```

Comment:
Will this be nil rather than an index exception (if that's a thing on Ruby)?

Reply:
You could do an assert using `include?` to test if `parameter_name` exists in the query string.

Reply:
Seems fine to me:

```